packages/core/src/internal/utils/auth.ts

```diff
@@ -108,6 +115,7 @@ export const decodeAuthToken = (token?: string): TokenAndEndpoints => {
         controlEndpoint: decodedLegacyToken.cp,
         cacheEndpoint: decodedLegacyToken.c,
         tokenEndpoint: decodedLegacyToken.c,
+        storageEndpoint: decodedLegacyToken.c,
```

Comment:
We need to decide what to do with handling legacy tokens for these new clients. This is going to give the wrong endpoint, so we'll either need to compute it from an existing claim or just declare that the storage client doesn't accept legacy tokens like the token client. The same is true of the `tokenEndpoint` already here, which should start with `token.`.

Reply:
we don't need to support legacy tokens for the new clients. if we already have a pattern for that in token client, yes, let's follow that pattern